visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -15,15 +16,12 @@ export class CodeChartaService {
 	public loadFiles(nameDataPairs: NameDataPair[]): Promise<void> {
 		return new Promise((resolve, reject) => {
 			nameDataPairs.forEach((nameDataPair: NameDataPair) => {
-				const validationResult: CCValidationResult = FileValidator.validate(nameDataPair.content)
-				if (validationResult.error.length === 0) {
-					const ccFile = this.getCCFile(nameDataPair.fileName, nameDataPair.content)
-					NodeDecorator.preDecorateFile(ccFile)
-					this.storeService.dispatch(addFile(ccFile))
-					if (validationResult.warning.length !== 0) {
-						reject(validationResult)
-					}
-				} else {
+				const validationResult: CCValidationResult = validate(nameDataPair.content)
```

Comment:
Ideally, the `validate` function would just throw in case of an error. If that's the case, the promise would immediately be rejected with the error that the function threw.

A validation function should ideally also not return anything. It's just there to synchronously validate that everything works as expected and throws an error if not.

In this specific case it would also have the benefit of stopping the forEach loop. Right now it will just continue to run, while with throwing an error it would stop right away.